crates/admin/src/cluster_controller/scheduler.rs

```diff
@@ -592,45 +588,36 @@ impl<T: TransportConnect> Scheduler<T> {
     /// Selects a leader based on the current target leader, observed cluster state and preferred leader.
     ///
     /// 1. Prefer worker nodes that are caught up
-    ///    2.1 Choose the current target leader
-    ///    2.2 Choose the preferred leader
-    ///    2.3 Pick any of the nodes in the current partition configuration
     /// 2. Pick worker nodes that are alive
-    fn select_leader(
-        &mut self,
-        partition_id: &PartitionId,
-        observed_cluster_state: &ObservedClusterState,
-    ) {
+    fn select_leader(&mut self, partition_id: &PartitionId, old_cluster_state: &OldClusterState) {
         let Some(partition) = self.partitions.get_mut(partition_id) else {
             return;
         };
 
-        if let Some(observed_partition_state) = observed_cluster_state.partition_state(partition_id)
-        {
+        TaskCenter::with_current(|tc| {
+            let cluster_state = tc.cluster_state();
```

Comment:
for the future. I think we can just keep cluster state in `self` instead of getting it from tc every time, just a thought.

Reply:
True. Will update it.